docs/source/federated-types/sharing-types.mdx

```diff
@@ -0,0 +1,404 @@
+
+Applying the `@shareable` directive to a type is equivalent to applying it to every _field_ of that type. You can also mark individual fields as `@shareable`, like so:
+
+<CodeColumns>
+
+```graphql{3}
+# Subgraph A
+type Food {
+  name: String! @shareable
+  cost: Int!
+}
+```
+
+```graphql{3}
+# Subgraph B
+type Food {
+  name: String!  @shareable
+  inStock: Boolean!
+}
+```
+
+<></>
+
+</CodeColumns>
+
+To learn how shared fields can differ across subgraphs, see [Varying shared fields](#varying-shared-fields).
+
+## Explicit and implicit type sharing
+
+### Objects
+
+By default in Federation 2, a single object field _can't_ be defined in more than one subgraph schema. To enable multiple subgraphs to define the same object field, you must first mark that field as `@shareable`.
+
+For example, let's say you'd like to define a `Product` type in two subgraphs:
+
+<p style="margin-bottom: 0">❌</p>
+
+<CodeColumns>
+
+```graphql{3, 4}
+# Subgraph A
+type Product {
+  name: String!
+  price: Int
+}
+```
+
+```graphql{3, 4}
+# Subgraph B
+type Product {
+  name: String!
+  price: Int
+}
+```
+
+</CodeColumns>
+
+Attempting to compose these two subgraph schemas together will [**break composition**](./composition/#breaking-composition). The gateway doesn't know _which subgraph_ is responsible for resolving the `Product` type's fields.
+
+If we want both Subgraph A _and_ Subgraph B to be able to resolve the fields of `Product`, we can explicitly mark those fields as [`@shareable`](#using-shareable):
+
+<p style="margin-bottom: 0">✅</p>
+
+<CodeColumns>
+
+```graphql{2}
+# Subgraph A
+type Product @shareable {
+  name: String!
+  price: Int
+}
+```
+
+```graphql{2}
+# Subgraph A
+type Product @shareable {
+  name: String!
+  price: Int
+}
+```
+
+</CodeColumns>
+
+Both subgraphs A and B can now resolve the `name` and `price` fields for the `Product` type, and our subgraph schema will successfully compose into a supergraph schema.
+
+> As an alternative, if you want Subgraphs A and B to resolve _different_ fields of `Product`, you can designate the `Product` type as an [entity](../entities). 
+
+### Enums, unions, and interfaces
+
+In Federation 2, `enum`, `interface`, and `union` type definitions can differ between subgraphs without explicit sharing:
+
+<CodeColumns>
+
+```graphql{2, 4, 10}
+# Subgraph A
+union Media = Book | Movie
+
+enum Color {
+  RED
+  GREEN
+  BLUE
+}
+
+interface User {
+  name: String!
+}
+```
+
+```graphql{2, 4, 10}
+# Subgraph B
+union Media = Book | Podcast
+
+enum Color {
+  CYAN
+  MAGENTA
+  YELLOW
+}
+
+interface User {
+  name: String!
+  age: Int!
+}
+```
+
+<></>
+
+</CodeColumns>
+
+Compositional logic merges these definitions in your supergraph schema:
+
+```graphql
+# Supergraph schema
+union Media = Book | Movie | Podcast
+
+enum Color {
+  RED
+  GREEN
+  BLUE
+  CYAN
+  MAGENTA
+  YELLOW
+}
+
+# The object types that implement this interface are
```

Comment:
I think the interface story is a _little_ more complicated than we currently describe here. I need to do some research 🤔

Reply:
Yeah...the main point I wanted to get across was that you can add/redefine fields in interfaces across multiple subgraphs without explicit sharing (and that's okay because interfaces don't _technically_ resolve their own fields). If the faux "supergraph schema" is confusing I can ask around and see if there is a better way to represent this?

Reply:
Totally, it's a good example to include! I'm just not sure whether there are extra concerns to highlight for interfaces, I need to reread some of the big design docs